velox/common/memory/SharedArbitrator.cpp

```diff
@@ -237,19 +166,136 @@ SharedArbitrator::~SharedArbitrator() {
   }
 }
 
-void SharedArbitrator::getCandidateStats(
+void SharedArbitrator::addPool(const std::shared_ptr<MemoryPool>& pool) {
+  VELOX_CHECK_EQ(pool->capacity(), 0);
+  VELOX_CHECK(pool->isRoot());
+  {
+    std::unique_lock guard{poolLock_};
+    VELOX_CHECK_EQ(candidates_.count(pool.get()), 0);
+    candidates_.emplace(pool.get(), pool);
+  }
+
+  std::lock_guard<std::mutex> l(stateLock_);
+  const int64_t maxBytesToReserve =
```

Comment:
Elsewhere we seem to be using uint64_t, but here local vars are signed.
Should we use uint64_t to be consistent?